### DISCUSSION TITLE
Fetch Data globally in Next Js 13

### DISCUSSION THREAD

**Discussion by datageek06** [vercel/next.js#45152] [Help]
Hi everyone.

How can I fetch data from API or headless CMS **once**, for example in app/page, and then use this data anywhere on the site?

Should I use getStaticProps?

PLEASE HELP!!

**Comment by guitheengineer** [comment]
You can fetch the data (where you think is best for your case, on the server or the client) and pass it through props or context.

There are also libraries like [swr](https://swr.vercel.app/docs/with-nextjs) and [tanstack query](https://tanstack.com/query/v4/docs/react/guides/ssr) where you can use a hook able to return the same response (cached) across your components.
